opssc-agent/src/chaincode-operator.ts

```diff
@@ -207,34 +207,41 @@ export class ChaincodeOperatorImpl implements ChaincodeOperator {
    * @returns {Promise<void>}
    */
   async download(): Promise<void> {
-    logger.info('[START] Download chaincode');
-    this.notifier?.notifyProgress('[START] Download chaincode');
+    try {
+      logger.info('[START] Download chaincode');
+      this.notifier?.notifyProgress('[START] Download chaincode');
 
-    const sourceAbsolutePath = this.sourceAbsolutePath();
-    const sourceParentAbsolutePath = path.resolve(sourceAbsolutePath, '..');
+      const sourceAbsolutePath = this.sourceAbsolutePath();
+      const sourceParentAbsolutePath = path.resolve(sourceAbsolutePath, '..');
 
-    fs.mkdirpSync(sourceParentAbsolutePath);
+      fs.mkdirpSync(sourceParentAbsolutePath);
 
-    if (fs.existsSync(sourceAbsolutePath)) {
-      logger.debug('Move old existing chaincode sourcecode to \'.orginal\'');
-      fs.removeSync(`${sourceAbsolutePath}.original`);
-      fs.moveSync(sourceAbsolutePath, `${sourceAbsolutePath}.original`);
-    }
+      if (fs.existsSync(sourceAbsolutePath)) {
+        logger.debug('Move old existing chaincode sourcecode to \'.orginal\'');
+        fs.removeSync(`${sourceAbsolutePath}.original`);
+        fs.moveSync(sourceAbsolutePath, `${sourceAbsolutePath}.original`);
+      }
 
-    let git = simplegit(sourceParentAbsolutePath);
-    const remote = this.remoteGitRepositoryURL(true);
-    await git.clone(remote);
-    git = simplegit(sourceAbsolutePath);
-    await git.checkout(this.proposal.chaincodePackage.commitID);
+      let git = simplegit(sourceParentAbsolutePath);
+      const remote = this.remoteGitRepositoryURL(true);
+      await git.clone(remote);
+      git = simplegit(sourceAbsolutePath);
+      await git.checkout(this.proposal.chaincodePackage.commitID);
 
-    // Build chaincode for typescript
-    if (this.proposal.chaincodePackage.type === 'typescript') {
-      execCommand('npm install', false, this.chaincodeAbsolutePath());
-      execCommand('npm run build', false, this.chaincodeAbsolutePath());
-    }
+      // Build chaincode for typescript
+      if (this.proposal.chaincodePackage.type === 'typescript') {
+        execCommand('npm install', false, this.chaincodeAbsolutePath());
+        execCommand('npm run build', false, this.chaincodeAbsolutePath());
+      }
 
-    logger.info('[END] Download chaincode');
-    this.notifier?.notifyProgress('[END] Download chaincode');
+      logger.info('[END] Download chaincode');
+      this.notifier?.notifyProgress('[END] Download chaincode');
+    } catch(e) {
+      if (this.config.gitUser && this.config.gitPassword) {
+        const maskedErrorMessage = (e.message as string).replace(this.config.gitUser, "*****").replace(this.config.gitPassword, "*****");
+        throw new Error(maskedErrorMessage);
+      }
```

Comment:
I guess throwing an Error is missing in the "else" case (i.e. gitUser or gitPassword is not set)